### ISSUE TITLE
Openstack - Resource agent to manage floating IPs

### ISSUE THREAD

**Issue by mgrzybek** [ClusterLabs/resource-agents#1125]
### Description
In a bare-metal environment, resource agents such as IPAddr2 are used to move addresses. Moreover, an AWS EC2 agent has been written.

In order to use the same mechanism with Openstack, new resource agents should be provided. For example:

1. set node attributes like instance ID, ports ID + subnets ID…
2. use these attributes to attach a floating IP to a port

### Proposal

1. resource agent named `os-info` to populate the CIB
2. resource agent named `os-floating-ip` to manage the resource

That could be a good starting point for other agents such as `os-cinder-volume`.

### Questions

Do you think these names are OK? Any suggestion?

**Comment by mbaldessari** [comment]
I am having a bit of trouble understanding the requirements here. Can you make some concrete example of the problems you're trying to solve?

**Comment by krig** [comment]
I'm also not sure I understand the purpose of the proposed agents... as a side note, I would prefer `openstack-*` as prefix rather than `os-`, since `OS` can be interpreted as "operating system". At least, that would be my first association.

**Comment by mgrzybek** [comment]
My IaaS team does not provide LBaaS service (using Neutron, Octavia…) or DBaaS.
As a consequence I need to deal with high-availability by myself. The corosync/pacemaker stack should be able to migrate an IP address or storage.

A "classic" resource group like:

```
Resource Group: g-lb
     haproxy		(systemd:haproxy):			Started node-1
     service-ip		(ocf::heartbeat:IPaddr2):		Started node-1

Clone Set: c-dlm-clvm [dlm-clvm]
     Started: [ node-1 node-2 node-3 ]

Resource Group: g-storage
     my-lvm-vg		(ocf::heartbeat:LVM):				Started node-2
     my-fs		(ocf::heartbeat:Filesystem):			Started node-2
```

Using the new RAs:

```
Clone Set: c-openstack-info [openstack-info]
     Started: [ node-1 node-2 node-3 ]

Resource Group: g-lb
     haproxy		(systemd:haproxy):			Started node-1
     service-ip		(ocf::heartbeat:openstack-floating-ip):	Started node-1

Resource Group: g-storage
     my-lun		(ocf::heartbeat:openstack-cinder-volume):	Started node-2
     my-lvm-vg		(ocf::heartbeat:LVM):				Started node-2
     my-fs		(ocf::heartbeat:Filesystem):			Started node-2


Node Attributes:
* Node node-1:
    + openstack_az                           	: paris-east
    + openstack_flavor                       	: c1.small  
    + openstack_id                           	: 60ac4343-5828-49b1-8aac-7c69b1417f31
    + openstack_ports                        	: 7960d889-9750-4160-bf41-c69a41ad72d9:96530d18-57a3-4718-af32-30f2a74c22a2
* Node node-2:
    + openstack_az                           	: paris-north
    + openstack_flavor                       	: c1.small  
    + openstack_id                           	: 60ac4853-5828-49b1-8bac-7c69b1417f31
    + openstack_ports                        	: 7960d889-9230-4160-bf41-c69a41ad72d9:96530d18-57a3-4718-af32-30f2a74d42a9
* Node node-2:
    + openstack_az                           	: paris-west
    + openstack_flavor                       	: c1.small  
    + openstack_id                           	: 60ac4853-6128-49b1-8bac-7c69b8977f31
    + openstack_ports                        	: 7960d889-9230-4160-bf09-c69a41ad72d9:96890d18-57a3-4718-af32-30f2a83d42b8
```

**Comment by mgrzybek** [comment]
This has been merged thanks to #1147
